arduinoClient/requestHandler.ino

```diff
@@ -0,0 +1,59 @@
+bool request_action(int action_index, char* response_buffer, int buffer_size){
```

Comment:
The fact that the global var _client_ is accessed from 2 files, 4 functions **and** needs to be manually manipulated for correct behaviour is very error-prone.
I would
A) pass it explicitly to all its functions from `loop` (rename it inside or outside so the name differs across scopes):
- set_finished
- request_action
- request_node_details
- do_request
- get_json_body
Make sure to pass it as a reference (`GSMClient&`).

B) put comments with expected pre and post-conditions right berfore those functions. E.g. 
- _Expects client to be open_ 
-  _Caller is expected to call `client.close` after the functions returns_.

This is not a perfect solution, but it will already clear up a lot of things for Those Who Come After.